src/test/resources/fixtures/hmh-ri/caliperEventGradeGraded.json

```diff
@@ -1,60 +1,54 @@
 {
   "@context": "http://purl.imsglobal.org/ctx/caliper/v1p1",
-  "id":"urn:uuid:uniqueIdPerEvent",
+  "id":"urn:uuid:a50ca17f-5971-47bb-8fca-4e6e6879001d",
   "type": "GradeEvent",
   "actor": {
-    "id": "https://url/to/lexileService",
+    "id": "https://www.hmhco.com/ReadingInventory",
```

Comment:
The actor id here should be the lexile service because that's the person/application that generated this event - can you confirm my thinking @padraig-ohiceadha?

Secondly, because Ed Analytics uses Actor id for its partition key, it means until this event has student user id as the actor id, it won't get processed correctly.... so my question for @padraig-ohiceadha what can we do in this scenario? Would it be ok to have actor as the student and 'scoredBy' as the lexile service https://github.com/hmhco/caliper-common-fixtures/pull/3/files/8031736dcab941210bf9e30217ddc4f2e218f4bd..ef6e9e8d2b3a3ffe3d96cb2e5fec1138365d313c#diff-cda1d6654d25effaed17317c5a5e8f44R40 ?

Reply:
Yes, the actor should be the Software Application which generated the event.
We do have flexibility in how granular we define applications, we could say everything is ED.
However as ED has more than one scoring service (e.g. Learnosity is also scoring, and we'll hopefully have QTI scoring in the future) and Common Patterns scores also (and is working on Caliper at the moment also), it would be useful to be able to see what particular scoring service was used to score this Attempt

Reply:
Unfortunately the student is not an actor here. Unless we have an edge case where they are scoring their own results.
The actor is that who/which performs the action which leads to the generated.
As most analyses of this event will be more interested in who the assignee is than who the actor is I suppose it makes sense to use the assignee as the partition key.
Remember though that when manual scoring comes in to play there will be two people involved and you may be interested in doing some analytics by scorer also.
There is a general pattern of events with 2 human actors (you often also get in communication and collaboration), it might be an idea to discuss as a team how best to handle such events

Reply:
Just to clarify terminology here - ReadingInventory is the Lexile Service.

Reply:
thanks 👍